### ISSUE TITLE
Dialog background colour prop

### ISSUE THREAD

**Issue by stephenpowers-sage** [Sage/carbon#6257]
### Desired behaviour

Currently, Dialog only supports a grey background variant, so this appears to be a misalignment with Product Design System which suggests we should be able to use a White one. 

This is a blocker to addressing an accessibility issue where the colour of the Stepper on a Dialog does not have the required colour contrast requirement. 

### Current behaviour

Only grey background on Dialog is available. 

### Suggested Solution

A new prop to select grey or white background. Perhaps backgroundGrey default true if set to false then white?

### CodeSandbox or Storybook URL

https://carbon.sage.com/?path=/docs/dialog--default-story

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by edleeks87** [comment]
@ljemmo would you mind reviewing this feature request please?

**Comment by ljemmo** [comment]
hi @edleeks87 by default, we'd like dialog, sidebar and drawer to have a white background: {colors.utility.yang.100}.

However,  if designers wish, they can switch on the gray background for those components using a new background-color prop. This would use the following token: {colors.utility.major.150}.

Specs below:
https://www.figma.com/file/Tau6qv5iyr8wthWNqYktej/DS-Official-Components?type=design&node-id=25410-300182&mode=design&t=XQ2vzrmAcr2zZOxJ-4

**Comment by tomdavies73** [comment]
FE-6191

**Comment by Parsium** [comment]
@harpalsingh do you happen to know if any products still require the grey background colour for `Dialog`? If not, we can just change the colour to the white colour token

**Comment by ljemmo** [comment]
@Parsium FYI i chatted with harpal about this when we did the work. Aparrently there are still some usecases where gray is needed (very complex dialogs with lots of cards and tiles on top). So it would unfortunately still be needed for now. However we do want white to be default over gray.

**Comment by harpalsingh** [comment]
@Parsium @ljemmo yep as me and Luke have discussed the reasons we still need to provide the grey option, so it should remain as something designers/developers can configre on the dialog

**Comment by Parsium** [comment]
Awesome, thanks for this @ljemmo @harpalsingh 👍🏼
